scss/common/_forms.scss

```diff
@@ -225,4 +225,89 @@
         right: 3px;
     }
 
+}
+
+@include exports("forms/integration-with-bootstrap-forms") {
+    $bootstrap-integration-spacer: 1rem !default;
+
+    .k-widget.form-control {
+        padding: 0;
+    }
+
+    .k-widget.form-control,
+    .k-maskedtextbox.form-control .k-textbox,
+    .k-textbox.form-control {
+        width: 100%;
+    }
+
+    .k-autocomplete.form-control .k-input {
+        box-sizing: border-box;
+    }
+
+    .input-group {
+        .k-dateinput-wrap,
+        .k-dropdown-wrap,
+        .k-multiselect-wrap,
+        .k-numeric-wrap,
+        .k-picker-wrap,
+        .k-maskedtextbox .k-textbox {
+            border-top-right-radius: 0;
+            border-bottom-right-radius: 0;
+
+            .k-select {
+                border-top-right-radius: 0;
+                border-bottom-right-radius: 0;
+            }
+        }
+
+        .k-widget.form-control:not(:first-child):not(:last-child) {
+            .k-dateinput-wrap,
+            .k-dropdown-wrap,
+            .k-multiselect-wrap,
+            .k-numeric-wrap,
+            .k-picker-wrap,
+            > .k-textbox {
+                border-radius: 0;
+            }
+        }
+    }
+
+    .input-group-addon + .k-widget.form-control {
+        .k-dateinput-wrap,
+        .k-dropdown-wrap,
+        .k-multiselect-wrap,
+        .k-numeric-wrap,
+        .k-picker-wrap,
+        > .k-textbox {
+            border-top-left-radius: 0;
+            border-bottom-left-radius: 0;
+            border-top-right-radius: $border-radius;
+            border-bottom-right-radius: $border-radius;
+        }
+    }
+
+    .k-widget.form-control .k-input,
+    .k-maskedtextbox.form-control .k-textbox,
+    .k-textbox.form-control {
+        padding: ($bootstrap-integration-spacer / 2) ($bootstrap-integration-spacer * .75);
+    }
+
+    $bootstrap-integration-input-height: $bootstrap-integration-spacer * 2.25 !default;
```

Comment:
Bootstrap uses the variables `$input-padding-y`, `$input-padding-x`, and `$input-line-height` to size form inputs. With this in mind, you can define the spacing in this file like this:

    $twbs-font-size-base: 1rem !default;
    $twbs-input-padding-x: .5rem !default;
    $twbs-input-padding-y: .75rem !default;
    $twbs-input-line-height: 1.25 !default;

(values are copied from Bootstrap to work in kendo-theme-default without introducing a dependency to Bootstrap).
With these declarations, you can write the following lines:

    padding: ($bootstrap-integration-spacer / 2) ($bootstrap-integration-spacer * .75);
    $bootstrap-integration-input-height: $bootstrap-integration-spacer * 2.25;

as

    padding: $twbs-input-padding-y $twbs-input-padding-x;
    $bootstrap-integration-input-height: $twbs-input-padding-y * 2 + $twbs-font-size-base * $twbs-input-line-height;

While a bit more complex, the above mimics the Bootstrap variables, and reacts to changes in them. If customers load the bootstrap SCSS and modify it (like in kendo-theme-bootstrap), the values can be defined as:

    $twbs-font-size-base: $font-size-base !default;
    $twbs-input-padding-x: $input-padding-x !default;
    $twbs-input-padding-y: $input-padding-y !default;
    $twbs-input-line-height: $input-line-height !default;

Thus inheriting the variable values from Bootstrap.